protoc-gen-twirp/generator.go

```diff
@@ -1334,6 +1334,13 @@ func (t *twirp) generateServerProtobufMethod(service *descriptor.ServiceDescript
 	t.P()
 	t.P(`  buf, err := `, t.pkgs["ioutil"], `.ReadAll(req.Body)`)
 	t.P(`  if err != nil {`)
+	t.P(`    if ctx.Err() == context.Canceled  {`)
+	t.P(`    		s.writeError(ctx, resp, twirp.NewError(twirp.Canceled, "context canceled"))`)
```

Comment:
Optional: It seems unnecessary to write out the meaning of the error three times here (checking equality, setting Twirp status, adding message), and another three times in the next block. What do you think about replacing the last one of those with a call to `context.Canceled.Error()`, either via `ctx.Err().Error()` (since we have just determined that the two are `==`), or via `if ctxErr := context.Canceled; ctx.Err() == cxtErr /* ... */ ctxErr.Error()`?

It's not critical to change, and I'm not thrilled with any of the options I've provided for changing it. But it seems like more repetition than we need.

Reply:
Done